File: src/main/java/jenkem/shared/data/ImageCss.java
package jenkem.shared.data;

import java.io.Serializable;
import javax.jdo.annotations.IdentityType;
import javax.jdo.annotations.PersistenceCapable;
import javax.jdo.annotations.PrimaryKey;

/**
 * Persistent CSS data for converted images.
 */
@PersistenceCapable(identityType = IdentityType.APPLICATION)
public class ImageCss implements ImagePartIfc, Serializable {
    private static final long serialVersionUID = 6300119792059722410L;

    @PrimaryKey
    private String _id; // mongodb identifier
    private String name;
    private String css;

    public ImageCss() {
    }

    public ImageCss(final String name, final String css) {
        this._id = name;
        this.name = name;
        this.css = css;
    }

    public final String getName() {
        return name;
    }

    public final String getCss() {
        return css;
    }

    @Override
    public final String toString() {
        return name;
    }
}

File: src/main/scala/jenkem/persistence/data/ImageCss.scala
/*
 * #%L
 * ImageCss.scala - Jenkem - Tok - 2012
 * %%
 * Copyright (C) 2012 - 2013 Lukas Steiger <lsteiger4@gmail.com>
 * %%
 * This program is free software. It comes without any warranty, to
 * the extent permitted by applicable law. You can redistribute it
 * and/or modify it under the terms of the Do What The Fuck You Want
 * To Public License, Version 2, as published by Sam Hocevar.
 * See http://www.wtfpl.net/ for more details.
 * #L%
 */
package jenkem.persistence.data

import javax.jdo.annotations.IdentityType
import javax.jdo.annotations.PersistenceCapable
import javax.jdo.annotations.PrimaryKey

/**
 * Persistent CSS data for converted images.
 */
@PersistenceCapable(identityType = IdentityType.APPLICATION)
class ImageCss(var name: String, var css: String) extends ImagePartTrait {
  @PrimaryKey var _id = name
  override def toString: String = name
}
